src/Lightweight/DataMapper/DataMapper.hpp

```diff
@@ -1219,6 +1227,15 @@ std::optional<Record> DataMapper::QuerySingle(PrimaryKeyTypes&&... primaryKeys)
     return { std::move(resultRecord) };
 }
 
+template <typename Record, typename... PrimaryKeyTypes>
+std::optional<Record> DataMapper::QuerySingle(PrimaryKeyTypes&&... primaryKeys)
+{
+    auto record = QuerySingleNoRelations<Record>(std::forward<PrimaryKeyTypes>(primaryKeys)...);
+    if (record)
+        ConfigureRelationAutoLoading(record.value());
```

Comment:
```suggestion
    if (record)
        ConfigureRelationAutoLoading(*record);
```
Since we just tested on the availability right one line above, just use deref-overload, as `.value()` would test again (and throw, in case, which won't happen here).